rohmu/object_storage/azure.py

```diff
@@ -253,21 +254,27 @@ def _parse_length_from_content_range(cls, content_range: str) -> int:
 
         return int(content_range.split(" ", 1)[1].split("/", 1)[1])
 
-    def _stream_blob(self, key: str, fileobj: BinaryIO, progress_callback: ProgressProportionCallbackType) -> None:
+    def _stream_blob(self, key: str, fileobj: BinaryIO, byte_range: Optional[tuple[int, int]], progress_callback: ProgressProportionCallbackType) -> None:
         """Streams contents of given key to given fileobj. Data is read sequentially in chunks
         without any seeks. This requires duplicating some functionality of the Azure SDK, which only
         allows reading entire blob into memory at once or returning data from random offsets"""
         file_size = None
-        start_range = 0
+        start_range = byte_range[0] if byte_range else 0
         chunk_size = self.conn._config.max_chunk_get_size  # type: ignore [attr-defined] # pylint: disable=protected-access
         end_range = chunk_size - 1
         blob = self.conn.get_blob_client(self.container_name, key)
         while True:
             try:
                 # pylint: disable=protected-access
-                download_stream = blob.download_blob(offset=start_range, length=chunk_size)
+                if byte_range:
```

Comment:
this should make sure byte_range[1] > byte_range[0]

Reply:
Probably yes. I only worked on the google part, the other implementations are from Markus... maybe I should give a look to ensure we have proper validation & tests for this for the other implementations too.

Reply:
I would put a helper and have the byte_range validated across all the drivers. At quite a few places the assumption needs to hold or things go wonky.